adapter/PreciceInterface.c

```diff
@@ -573,10 +573,10 @@ void PreciceInterface_NodeConnectivity(PreciceInterface *interface, SimulationDa
   PreciceInterface_ConfigureTetraFaces(interface, sim);
 }
 
-void PreciceInterface_EnsureValidNodesMeshID(PreciceInterface *interface)
+void PreciceInterface_EnsureValidNodesMeshID(PreciceInterface *interface, const char *type)
 {
   if (interface->nodesMeshID < 0) {
-    printf("Nodes mesh not provided in YAML config file\n");
+    printf("Nodes mesh not provided in YAML config file. They are required for writing/reading the data %s.\n", type);
```

Comment:
I am wondering a bit about the action call: should the user change something in the adapter config file, or in the mesh itself? Maybe refering to the documentation would help.

Reply:
Actually the original sentence is a bit misleading, as the problem is not *providing* the mesh, it's configuring correctly the YAML file. Given a mesh, either we configure the adapter to build a preCICE mesh from the nodes, or from the face centers. We can do both by configuring two interfaces in the YAML file (and creating two different preCICE meshes)

A different formulation would probably be more suited. Something like "This interface was configured to use nodes/faces, but it must be configured to use faces/nodes to transfer data Y". I'm open to suggestions.

To trigger the error, simply take any valid CalculiX test case and, in the `config.yaml` file, replace `faces-mesh` by `nodes-mesh` or vice-versa. (`mesh` seems to be an alias for `faces-mesh` and there's another one for connectivity, see https://github.com/precice/calculix-adapter/issues/87).
I first noticed this because it is actually impossible to do CHT with Dirichlet-Neumann coupling on the same mesh, since one requires nodes and the other one requires faces.

Reply:
I like your suggestion, let's go with that! :+1: Maybe even include some information about which "this interface" is?